src/jit/compiler.h

```diff
@@ -7729,7 +7729,19 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     // The minimum and maximum possible number of bytes in a SIMD vector.
     unsigned int maxSIMDStructBytes()
     {
+#if FEATURE_HW_INTRINSICS && defined(_TARGET_XARCH_)
```

Comment:
This could be confusing, so I would suggest adding a comment to the effect that in the case of AVX without AVX2, `Vector<T>` will be limited to 16 bytes, even though the full 32 byte AVX register length is supported for the HW intrinsics.